api/tests/unit/features/test_unit_feature_external_resources_views.py

```diff
@@ -813,3 +851,37 @@ def test_create_feature_external_resource_on_environment_with_v2(
         },
         timeout=10,
     )
+
+
+def test_cannot_create_feature_external_resource_for_the_same_feature_and_resource_uri(
+    admin_client_new: APIClient,
+    feature: Feature,
+    project: Project,
+    github_configuration: GithubConfiguration,
+    github_repository: GithubRepository,
+    feature_external_resource_gh_pr: FeatureExternalResource,
+) -> None:
+    # Given
+    feature_external_resource_data = {
+        "type": "GITHUB_PR",
+        "url": "https://github.com/repositoryownertest/repositorynametest/pull/1",
+        "feature": feature.id,
+        "metadata": {"state": "open"},
+    }
+
+    url = reverse(
+        "api-v1:projects:feature-external-resources-list",
+        kwargs={"project_pk": project.id, "feature_pk": feature.id},
+    )
+
+    # When
+    response = admin_client_new.post(
+        url, data=feature_external_resource_data, format="json"
+    )
+
+    # Then
+    assert response.status_code == status.HTTP_400_BAD_REQUEST
+    assert (
+        response.json()["non_field_errors"][0]
+        == "The fields feature, url must make a unique set."
+    )
```

Comment:
I don't understand why the fields feature and url do not make a unique set. Does the test set these as duplicates elsewhere?

Reply:
This is because already exists in this fixture:
this:https://github.com/Flagsmith/flagsmith/blob/e9e1e5939a7849157804da23059eb471bfaecc6b/api/conftest.py#L1044